chainerx_cc/chainerx/cuda/numeric.cuh

```diff
@@ -43,6 +43,16 @@ __device__ inline float Arccosh(float x) { return std::acoshf(x); }
 
 __device__ inline cuda::Float16 Arccosh(cuda::Float16 x) { return cuda::Float16{std::acoshf(static_cast<float>(x))}; }
 
+template <typename T>
+__device__ inline T Sign(T x) {
+    return IsNan(x) ? x : static_cast<T>((T{0} < x) - (x < T{0}));
```

Comment:
Perhaps this line should also be fixed similarly to `cuda::Float16` specialization
(Avoid implicit cast from `bool` to `int`)?

Reply:
Makes sense. On it.